### PR TITLE
Change default column behavior (Structure tab) - fix #1163

### LINKED ISSUE #1163: Unable to create text column without default value
- Sequel Ace Version: 3.3.3 (3034)
- Sequel Ace Source: App Store
- macOS Version: 10.15.7
- Processor Type: Intel
- MySQL Version: MariaDB 10.3.27
- macOS Localization: EN

**Is Issue Present in [Latest Beta](https://github.com/Sequel-Ace/Sequel-Ace/releases)?**
Yes, 3.4.0 (3035) Beta 1 same result

**Description**
When I create new column with text type (`CHAR`, `VARCHAR`, `TEXT`, etc.), here is no way to create it without `DEFAULT VALUE` set to empty string or NULL.

**Steps To Reproduce**
1. Open or create any table.
2. Switch to `Structure` tab.
3.  Click to `+` sign to add new Column.
4. Fill `Field`.
5. Change `Type` to `VARCHAR`.
6. Define `Length` to any number.
7. Press `Enter`.

Now is created Column with `DEFAULT NULL`.

**Steps To Reproduce**
1. Open or create any table.
2. Switch to `Structure` tab.
3.  Click to `+` sign to add new Column.
4. Fill `Field`.
5. Change `Type` to `VARCHAR`.
6. Define `Length` to any number.
7. Uncheck `Allow Null`
8. Press `Enter`.

Now is created Column with `DEFAULT ''`.

Here is no way to create column without default value.

**Expected Behaviour**
No default value until user explicitly fill it. Maybe add any way (small `×` button to `Default` parameter field, or at least right-click) option to remove default value from column.

**Related Issues**
Searched, nut no related issue found.

**Additional Context**
image

### PR DESCRIPTION
**Pull request by gduh**
## Changes:
- Change 'default' column behavior (Structure tab)

## Tested:
- Processors:
  - [X] Intel
  - [ ] Apple Silicon
- macOS Versions:
  - [ ] 10.12.x (Sierra)
  - [ ] 10.13.x (High Sierra)
  - [ ] 10.14.x (Mojave)
  - [] 10.15.x (Catalina)
  - [X] 11.x (Big Sur)
- Localizations:
  - [X] English
  - [ ] Spanish
  - [ ] Other (please specify)
- Xcode Version: 13 (13A233)

### CODE REVIEW

File: Source/Controllers/Preferences/SPPreferencesUpgrade.m [hunk 1]
```diff
@@ -76,6 +76,11 @@ void SPApplyRevisionChanges(void)
 		[importantUpdateNotes addObject:NSLocalizedString(@"There is a new option in Preferences->Alerts & Logs: \"Show warning before executing a query\". When enabled, you will be prompted to confirm that you want to execute an SQL query or edit a row.", @"Short important release note for new option in Preferences->Alerts & Logs")];
 	}
 
+    // Release note for 'Default' column behavior
+    if (recordedVersionNumber > 3043) {
+        [importantUpdateNotes addObject:NSLocalizedString(@"The use of the \"default\" column has changed since the last version of Sequel ACE:\n\n- No default value : Leave it blank.\n- String value : Use single '' or double quotes \"\" if you want an empty string or to wrap a string\n- Expression : Use parentheses (). Except for the TIMESTAMP and DATETIME columns where you can specify the CURRENT_TIMESTAMP function without enclosing parentheses.", @"Warning message about the default value")];
```
Comment (Kaspik) [inline Source/Controllers/Preferences/SPPreferencesUpgrade.m:81]: This needs to go to localized strings then?

**Comment by Jason-Morcos** [comment]
@gduh Can you resolve the conflict that's holding this from merge? Other than the conflict, do you feel like it's ready to go for the next release? Heavily tested?

**Comment by gduh** [comment]
Merge conflict resolved.

About the release note displayed at startup, I changed the `recordedVersionNumber` of the test to 20025. Hope that's ok.

    if (recordedVersionNumber > 20025) {
        ...
    }

On my side, I think the change is ok to go for the next release.

**Review by Jason-Morcos — APPROVED** [review]
Looks great to me!

**Pull request merged**